app/src/main/java/com/kylecorry/trail_sense/astronomy/ui/AstroChart.kt

```diff
@@ -141,53 +130,44 @@ class AstroChart(chart: Chart, private val onImageClick: () -> Unit) {
     }
 
     fun moveSun(position: Reading<Float>?) {
-        sunImage.data = if (position == null) {
-            emptyList()
-        } else {
-            Chart.getDataFromReadings(listOf(position), startTime) { it }
-        }
+        sunImage.data = position?.let { Chart.getDataFromReadings(listOf(it), startTime) { it } } ?: emptyList()
         updateSunArea()
     }
 
     fun setMoonImage(@DrawableRes icon: Int) {
-        moonImage.bitmap = bitmapLoader.load(icon, imageSize.toInt())
         if (icon != previousMoonImage) {
             bitmapLoader.unload(previousMoonImage)
+            previousMoonImage = icon
         }
-        previousMoonImage = icon
+        moonImage.bitmap = bitmapLoader.load(icon, imageSize.toInt())
```

Comment:
What is the reason this line was moved to below the if statement? Was there a bug you discovered?